src/support/file.cpp

```diff
@@ -27,8 +27,13 @@ T wasm::read_file(const std::string &filename, bool debug) {
     exit(EXIT_FAILURE);
   }
   infile.seekg(0, std::ios::end);
-  size_t insize = infile.tellg();
-  T input(insize + 1, '\0');
+  std::streamoff insize = infile.tellg();
+  if (sizeof(size_t) == 4 && insize >= 0xFFFFFFFFU) {
```

Comment:
Followed a code pattern in existing location in code, changed that to use numeric_limits as well.